project/PIPBuildPlugin.scala

```diff
@@ -0,0 +1,117 @@
+    val src = (Compile / pythonSource).value
+    val dest = (Python / target).value
+    if (!dest.exists()) dest.mkdirs()
+    s.log(s"Copying '$src' to '$dest'")
+    IO.copyDirectory(src, dest)
+    dest
+  }
+
+  /** Builds python distribution. */
+  val pyDist = Def.task {
+    val buildDir = (Python / target).value
+    val pyDist = (packageBin / artifactPath).value
+    val retcode = pySetup.toTask(" bdist_wheel").value
+    if(retcode != 0) throw new RuntimeException(s"'python setup.py' returned $retcode")
+    val results = IO.listFiles(buildDir / "dist", "*.whl")
+    require(results.length == 1, s"Expected to find single '.whl' file in ${buildDir / "dist"}")
+    IO.copyFile(results.head, pyDist)
+    pyDist
+  }
+
+  override def projectConfigurations: Seq[Configuration] = Seq(Python)
+
+  override def projectSettings = Seq(
+    assembly / test := {},
+    // Python command path
+    pythonCommand := "python",
+    // Run 'python setup.py' as in input task with arguments
+    pySetup := {
+      val s = streams.value
+      val wd = copyPySources.value
+      val args = spaceDelimited("<args>").parsed
+      val cmd = Seq(pythonCommand.value, "setup.py") ++ args
+      val ver = version.value
+      s.log.info(s"Running '${cmd.mkString(" ")}' in $wd")
+      Process(cmd, wd, "RASTERFRAMES_VERSION" -> ver).!
+    },
+    // Set the default path to python module sources
+    Compile / pythonSource := (Compile / sourceDirectory).value / "python",
+    // Register python directory as a source directory
+    Python / unmanagedSourceDirectories := Seq((Compile / pythonSource).value),
+    // Set the default path to python test sources
+    Test / pythonSource := (Test / sourceDirectory).value / "python",
+    // Add python test directory as a test source directory
+    Python / unmanagedSourceDirectories := Seq((Test / pythonSource).value),
+    // Add .py as a recognized python source extension
+    unmanagedSources / includeFilter := (unmanagedSources / includeFilter).value ||
+      (GlobFilter("*.py") | "*.cfg" | "*.in") -- DirectoryFilter,
+    Compile / `package` := (Compile / `package`).dependsOn(Python / packageBin).value,
+    Compile / packageBin / artifactPath := {
+      val dir = (Python / target).value
+      val art = (packageBin / artifact).value
+      val fileName = (Compile / packageBin / artifactPath).value.getName
+      dir / art.name / fileName
+    }
+  ) ++
+    inConfig(Python)(Seq(
+      target := target.value / "python",
+      packageBin := Def.sequential(Compile / packageBin, pyDist).value,
+      packageBin / artifact := {
+        val java = (Compile / packageBin / artifact).value
+        java.withType("zip").withClassifier(Some("python")).withExtension("zip")
+      },
+      packageBin / artifactPath := {
+        val art = (packageBin / artifact).value
+        val dest = (Compile / target).value
+        val ver = version.value
+        dest / s"${art.name}-python-$ver.zip"
+      },
+      test := {
```

Comment:
This doesn't yet work. Needs to set up the pyspark jar paths and test directories.